src/audio/codec_adapter/codec_adapter.c

```diff
@@ -522,52 +518,28 @@ static int codec_adapter_set_params(struct comp_dev *dev, struct sof_ipc_ctrl_da
 	ret = memcpy_s(dst, size - offset, src, cdata->num_elems);
 	assert(!ret);
 
-	/* Config has been copied now we can load & apply it depending on
-	 * codec state.
-	 */
-	if (!cdata->elems_remaining) {
-		switch (type) {
-		case CODEC_CFG_SETUP:
-			ret = load_setup_config(dev, codec->runtime_params, size);
-			if (ret)
-				comp_err(dev, "codec_adapter_set_params(): error %d: load of setup config failed.",
-					 ret);
-			else
-				comp_dbg(dev, "codec_adapter_set_params() load of setup config done.");
-
-			break;
-		case CODEC_CFG_RUNTIME:
-			ret = codec_load_config(dev, codec->runtime_params, size,
-						CODEC_CFG_RUNTIME);
-			if (ret) {
-				comp_err(dev, "codec_adapter_set_params() error %d: load of runtime config failed.",
-					 ret);
-				break;
-			} else {
-				comp_dbg(dev, "codec_adapter_set_params() load of runtime config done.");
-			}
-
-			if (codec->state >= CODEC_INITIALIZED) {
-				/* We are already prepared so we can apply runtime
-				 * config right away.
-				 */
-				ret = codec_apply_runtime_config(dev);
-				if (ret)
-					comp_err(dev, "codec_adapter_set_params() error %x: codec runtime config apply failed",
-						 ret);
-				else
-					comp_dbg(dev, "codec_adapter_set_params() apply of runtime config done.");
-			} else {
-				cd->codec.r_cfg.avail = true;
-			}
-
-			break;
-		default:
-			comp_err(dev, "codec_adapter_set_params(): error: unknown config type.");
-			break;
-		}
-	} else {
+	/* return as more fragments of config data expected */
+	if (cdata->elems_remaining)
 		return 0;
+
+	/* config fully copied, now load it */
+	ret = load_setup_config(dev, codec->runtime_params, size);
```

Comment:
I didn't fully verify this, but I guess it's a correct way to merge the above two cases and an additional call to `codec_load_config()` isn't needed because it's already called inside `load_setup_config()`

Reply:
@lyakh looking more into this I realized that I could drop load_setup_config() altogether and modify module_load_config() to handle both runtime and setup configs. Updated accordingly.